lib/index.js

```diff
@@ -123,7 +127,7 @@ class PackWalker extends IgnoreWalker {
         // then we know path is a workspace directory. in order to not drop ignore rules
         // from directories between the workspaces root (prefix) and the workspace itself
         // (path) we need to find and read those now
-        const relpath = relative(options.prefix, dirname(options.path))
```

Comment:
Why is `dirname` being removed here?

Reply:
dirname was removing last directory and resulting into one directory behind..
if path = root/level1/level2
dirname(path) = root/level1 due to this ignore files present in level2 is never got considered.

to consider level2 files, this is removed, although i had tested by keeping files at each level but let me know if i am missing anything

Reply:
Given the state of the existing tests in this I'm very hesitant to remove something like this without fully understanding the side effects.  We've had several bugs here in the last few months that weren't caught due to tests.  If this is fixing a bug, we'd want to find out what that bug is and clearly define it in a test.  If it's introducing a bug we are obviously not currently capturing it in a test and would want to make sure we add one to prevent this kind of regressions.

Reply:
Definitely we need to add more test case to cover both regression and the bug.. the current bug is captured in the test written. I will analyze more to understand why dirname was included at first place and if some test is missing to cover that.

Reply:
I don't see test cases covering ignore files at 3 level deep, its included now. Please confirm if we are good to proceed.

Reply:
The reason we're using `dirname` is because this code path is in a workspace.  The comment above this line explains it.  Removing `dirname` means we're no longer addressing the situation described in the comment.

Reply:
what i understood from comment is we want to read directories present in between (a)PREFIX and (a/b/c)WORKSPACE.. but in that case the ignore file present in the workspace(a/b/c) is never considered..

if we pass workspace(a/b/c) without dirname, recursion in "readOutOfTreeIgnoreFiles" is doing job to read files present in each subdirectories(a/b/c and a/b and a) starting from workspace directory.